### ISSUE TITLE
[Bug] Encounter error in Step 5.2 in pod-security.md due to ray-ml docker images

### ISSUE THREAD

**Issue by Yicheng-Lu-llll** [ray-project/kuberay#873]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

When following step 5.2 in [pod-security.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/pod-security.md), a type error may occur after running  `python3 samples/xgboost_example.py`:
```
TypeError: Descriptors can not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```

The root cause is ray-ml docker images(pod-security has no relationship with this error). Ray has been updated to support the latest protobuf but tensorboardx 2.5.1 requires protobuf<=3.20.1,>=3.8.0. ray-ml docker images use the latest protobuf and thus cause the above error. 

If following the [ray doc](https://docs.ray.io/en/latest/ray-overview/installation.html) and installing the Ray and dependencies for Ray AI Runtime by running `pip install -U "ray[air]"`, protobuf version will be 3.19 instead of 4.21.10(the latest) because of pip's dependency check. 

But again, ray-ml docker images use the latest protobuf. There is already a similar open issue on the ray side: https://github.com/ray-project/ray/issues/31309.

The current solution is to downgrade the protobuf package:
`pip install protobuf==3.20.*`

### Reproduction script

See  [pod-security.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/pod-security.md)

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by kevin85421** [comment]
This bug is caused by Ray's docker image.

cc @DmitriGekhtman @juliusvonkohout

**Comment by kevin85421** [comment]
https://github.com/ray-project/ray/issues/31309#issuecomment-1407226449 is updated. cc @Yicheng-Lu-llll

**Comment by Yicheng-Lu-llll** [comment]
protobuf is downgraded to 3.20.3 in ray-ml docker images(see https://github.com/ray-project/ray/issues/31309#issuecomment-1407226449). I have rerun and everything works fine. 

![image](https://user-images.githubusercontent.com/51814063/215634026-f6aa6b22-b82f-45a1-bd80-c7e75afc73da.png)

**Comment by kevin85421** [comment]
This issue happens again in Ray 2.3.0. See https://github.com/ray-project/kuberay/pull/999 for more details.

**Comment by kevin85421** [comment]
This issue is fixed by #999 again.
